### PR TITLE
FIX,ENH: upgraded basebox to xenial, updated cpac_install to reflect …

### LINKED ISSUE #25: trusty version not showing mount correctly
## Setup:
Singularity container on CENTOS 7 server as the entry to a HPC cluster.

## Behavior:
```
singularity shell jdkent_cpac_latest-2018-01-24-be00acf97a2f.img 
Singularity: Invoking an interactive shell within container...

Singularity.jdkent_cpac_latest-2018-01-24-be00acf97a2f.img> $ python
Python 2.7.13 |Continuum Analytics, Inc.| (default, Dec 20 2016, 23:09:15) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import nipype
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/__init__.py", line 49, in <module>
    from .pipeline import Node, MapNode, JoinNode, Workflow
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/pipeline/__init__.py", line 10, in <module>
    from .engine import Node, MapNode, JoinNode, Workflow
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/pipeline/engine/__init__.py", line 12, in <module>
    from .workflows import Workflow
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/pipeline/engine/workflows.py", line 41, in <module>
    from ...interfaces.base import (traits, InputMultiPath, CommandLine,
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/interfaces/__init__.py", line 12, in <module>
    from .io import DataGrabber, DataSink, SelectFiles
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/interfaces/io.py", line 38, in <module>
    from ..utils.filemanip import copyfile, list_to_filename, filename_to_list
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/utils/filemanip.py", line 266, in <module>
    _cifs_table = _generate_cifs_table()
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/utils/filemanip.py", line 259, in _generate_cifs_table
    reverse=True)
  File "/usr/local/bin/miniconda/lib/python2.7/site-packages/nipype/utils/filemanip.py", line 258, in <lambda>
    key=lambda x: len(x[0]),
IndexError: list index out of range
```
*The error is coming from the `mount` command*

### Example
```
Singularity.jdkent_cpac_latest-2018-01-24-be00acf97a2f.img> $ mount
singularity on / type rootfs (rw)

mount: warning: /etc/mtab is not writable (e.g. read-only filesystem).
       It's possible that information reported by mount(8) is not
       up to date. For actual information about system mount points
       check the /proc/mounts file.
```
While the input expected is just `singularity on / type rootfs (rw)`: [see nipype code](https://github.com/nipy/nipype/blob/0.13.1/nipype/utils/filemanip.py#L257)

### PR DESCRIPTION
**Pull request by jdkent**
…CPAC, updated version to reflect the version of CPAC this repository is based on.

### Purpose
Fixes #25 

### Changes
- Dockerfile box from neurodebian's trusty to xenial
- cpac_install.sh was copied from [CPAC 1.0.2_disable_log](https://github.com/FCP-INDI/C-PAC/blob/v1.0.2_disable_log/scripts/cpac_install.sh)
- updated version in version file to reflect the change
- removed minimal afni (it wasn't compatible with xenial, so switched to downloading from neurodebian for ease)

good idea chrisfilo, ping @ccraddock

### CODE REVIEW

**Comment by chrisgorgo** [comment]
Pinging @ccradock

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ ENV FSLBROWSER /etc/alternatives/x-www-browser
 ENV ANTSPATH /opt/ants/bin/
 ENV DYLD_FALLBACK_LIBRARY_PATH /opt/afni
 ENV LD_LIBRARY_PATH /usr/lib/fsl/5.0:${LD_LIBRARY_PATH}
-ENV PATH /code:/opt/c3d/bin:/opt/ants/bin:/opt/afni:${FSLDIR}/bin:/usr/local/bin/miniconda/bin:${PATH}
+ENV PATH /code:/opt/c3d/bin:/opt/ants/bin:/opt/afni:${FSLDIR}/bin:/usr/local/bin/miniconda/envs/cpac/bin:${PATH}
```
Comment (jdkent) [inline Dockerfile]: I still need to fix this path to reflect that afni is now in `/usr/lib/afni/bin`

File: Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ ENV FSLBROWSER /etc/alternatives/x-www-browser
 ENV ANTSPATH /opt/ants/bin/
```
Comment (jdkent) [inline Dockerfile]: Should delete this, and other references to `/opt/afni`

**Review by jdkent — COMMENTED** [review]
I still need to clean up some potential redundancies and and fix/delete ENV names

**Comment by jdkent** [comment]
To encourage not recreated the wheel (or the environment in this case), I pushed most of the building of the dockerfile to cpac_install.sh, I'll submit a pull request on the main CPAC repository to keep the cpac_install.sh files equivalent.

The only frustrating thing is that I have to build the environment at runtime, but I think this will simplify maintenance of the BIDS-App to keep in step with the main CPAC repository, so in this scenario, the benefits may outweigh the risks. 

I still need to test this container with actual data to test if it works.

Comments welcome!

**Comment by ccraddock** [comment]
I don't have a lot of time now, will look more closely later, but have a comment now. We have had trouble in the past with how NeuroDebian builds AFNI where some of the applications, for example 3dSkullStrip, fail to work, complaining that they need to open a display. Also, CPAC doesn't need the full AFNI, or FSL to execute. If possible I would like to try to find a solution that allows us to install just parts of these tools. I agree that the afni_minimal way isn't the best.  But hopefully we can find another way ...

Also I would like to get away from cpac_setup.py. It defeats the dependency change detection mechanism of docker builds, and results in re-builds taking longer than they really should need to. I am sure that you have experienced this quite a bit. 

Also would like to make it so that the user doesn't run as root, messes up permissions for generated files. 

Thank you very much for help with maintaining the CPAC BIDS app!

**Comment by jdkent** [comment]
> We have had trouble in the past with how NeuroDebian builds AFNI where some of the applications, for example 3dSkullStrip, fail to work, complaining that they need to open a display.

RE: 3dSkullstrip, I did run into that, which is found in version 16.2, but appears to be fixed in 18. You can see a similar conversation in [mriqc](https://github.com/poldracklab/mriqc/pull/669).

> Also, CPAC doesn't need the full AFNI, or FSL to execute. If possible I would like to try to find a solution that allows us to install just parts of these tools. 

I agree, extra bloat isn't great,  again in line with with mriqc, perhaps osf versions of utilities needed could be uploaded and installed here, moving some of the installation/configuration outside of cpac_install.sh which in my mind is a con, but I think the reduced size is worth it, but I'm not comfortable with knowing exactly which utilities are needed from both packages. 
EDIT: the container is too big to even pass automated tests.

> Also I would like to get away from cpac_setup.py. It defeats the dependency change detection mechanism of docker builds, and results in re-builds taking longer than they really should need to. I am sure that you have experienced this quite a bit.

Ahh, this is what I was trying to avoid earlier (assuming you meant cpac_install.sh), since I thought the dependencies were made "just so" in the cpac_install.sh and I thought it would be difficult to keep track of the dependencies needed in two separate places (Dockerfile and cpac_install.sh). I agree a nice feature of the Dockerfile is being able to separate out the steps to reduce build time...
Now I see it as being better just using the dockerfile as you said, I think my confusion when I was changing the dockerfile was why there was a hybrid approach, and I wanted building to mostly happen in one environment.

> Also would like to make it so that the user doesn't run as root, messes up permissions for generated files.

Is this possible within docker? I don't know the ins and outs of docker, but I thought you could only run containers as root.

> BTW thank you very much for help with maintaining the CPAC BIDS app!

No Problem! Open source only works if we help.

**Comment by ccraddock** [comment]
Looks like the installation script is too effusive for travisci, the build failed with"

"The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).

The job has been terminated"

**Comment by ccraddock** [comment]
Yes, I did mean cpac_install.sh, it looks like you did a lot of good work to it. I will use it to update the install script in the main CPAC repository, that way it will still go to good use if we decide to go to a Dockerfile only build.

**Comment by jdkent** [comment]
Error message from travis-ci:
```
(9/9) Moving the image to the output folder...
 10,322,182,176 100%   36.28MB/s    0:04:31 (xfr#1, to-chk=0/1)
rsync: write failed on "/output/bids_cpac-2018-01-26-66919f7d2caa.img": No space left on device (28)
rsync error: error in file IO (code 11) at receiver.c(393) [receiver=3.1.2]
The command "docker run -ti --rm --privileged -v /var/run/docker.sock:/var/run/docker.sock -v ${HOME}/singularity_images:/output filo/docker2singularity "bids/cpac"" failed and exited with 11 during .
```

**Comment by chrisgorgo** [comment]
a) try `singularityware/docker2singularity` b) check storage on your destination hard drive c) check if your virtual machine (you are using a mac?) did not run out of space; if in doubt reset docker

**Comment by jdkent** [comment]
thanks @chrisfilo, I'm still wet behind the ears when it comes to travis, how do I determine how much disk space travis has when it's testing my build?

**Comment by chrisgorgo** [comment]
oh - that's the travis error - it might mean this image is too big to convert on travis

**Comment by ccraddock** [comment]
Oh yeah, that’s one of the reasons we reduced AFNI. Instead of using neurodebian, can we install from the official AFNI tarball but only extract the required files?

**Comment by jdkent** [comment]
woo! it passed, ready for comments again, 
Should I include cpac_templates.tar.gz in the container? or was there a url it was downloaded from?

**Pull request merged**